src/backend/joanie/demo/management/commands/create_demo.py

```diff
@@ -0,0 +1,358 @@
+            print(f" {elapsed_time:g} seconds")
+
+        self.__class__.total_time += elapsed_time
+        return elapsed_time
+
+
+def create_free_enrollments(queue):
+    """
+    Create random enrollments in bulk.
+    The number of enrollments actually created may vary due to conflicts that we decide
+    to not handle because we only care about the order of magnitude of objects created.
+    """
+    state_choices = [s[0] for s in enums.ENROLLMENT_STATE_CHOICES]
+    course_runs_ids = models.CourseRun.objects.values_list("id", flat=True)
+    users_ids = models.User.objects.values_list("id", flat=True)
+    nb_enrollments = (
+        defaults.NB_OBJECTS["enrollments"] - models.Enrollment.objects.count()
+    )
+    for _i in range(nb_enrollments):
+        queue.push(
+            models.Enrollment(
+                user_id=random.choice(users_ids),  # nosec
+                course_run_id=random.choice(course_runs_ids),  # nosec
+                is_active=random_true_with_probability(0.7),
+                state=random.choice(state_choices),  # nosec
+                was_created_by_order=False,
+            )
+        )
+    queue.flush()
+
+
+# pylint: disable=too-many-branches,too-many-locals,too-many-statements
+def create_demo():
+    """
+    Create a database with demo data for developers to work in a realistic environment.
+    """
+    site = Site.objects.get(id=1)
+    site.domain = getattr(settings, "JOANIE_DEMO_DOMAIN", defaults.DEFAULT_DEMO_DOMAIN)
+    site.name = "Joanie demonstration"
+    site.save()
+
+    queue = BulkQueue()
+
+    with Timeit("Creating users"):
+        for i in range(defaults.NB_OBJECTS["users"]):
+            queue.push(
+                models.User(
+                    username=f"user{i:d}",
+                    email=f"user{i:d}@openfun.fr",
+                    password="!",  # nosec
+                    is_superuser=False,
+                    is_active=True,
+                    is_staff=False,
+                    language="fr-fr",
+                    first_name=f"Firstname {i:d}",
+                    last_name=f"Lastname {i:d}",
+                )
+            )
+        queue.flush()
+
+    with Timeit("Creating organizations"):
+        for _i in range(defaults.NB_OBJECTS["organizations"]):
+            queue.push(factories.OrganizationFactory.build())
+        queue.flush()
+
+    with Timeit("Creating courses"):
+        for _ in range(defaults.NB_OBJECTS["courses"]):
+            queue.push(factories.CourseFactory.build())
+        queue.flush()
+
+    with Timeit("Creating certificate definitions"):
+        for _ in range(defaults.NB_OBJECTS["products"]):
+            queue.push(factories.CertificateDefinitionFactory.build())
+        queue.flush()
+
+    with Timeit("Creating products"):
+        for certificate_definition in models.CertificateDefinition.objects.only(
+            "id"
+        ).iterator():
+            queue.push(
+                factories.ProductFactory.build(
```

Comment:
Currently, it appears there are some issues when building models with translated fields.

E.g: In the case of Product, title is not defined so the admin change view can be accessed...

Reply:
This is now fixed. I didn't use bulk creates for objects that have translated fields. They are not created in huge quantities so this is ok.